compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs

```diff
@@ -112,8 +115,25 @@ struct CoverageMapGenerator {
 }
 
 impl CoverageMapGenerator {
-    fn new() -> Self {
-        Self { filenames: FxIndexSet::default() }
+    fn new(tcx: TyCtxt<'_>, version: u32) -> Self {
+        let mut filenames = FxIndexSet::default();
+        if version >= 5 {
+            // LLVM Coverage Mapping Format version 6 (zero-based encoded as 5)
+            // requires setting the first filename to the compilation directory.
+            // Since rustc generates coverage maps with relative paths, the
+            // compilation directory can be combined with the the relative paths
+            // to get absolute paths, if needed.
+            let working_dir = tcx
+                .sess
+                .opts
+                .working_dir
+                .to_string_lossy(FileNameDisplayPreference::Remapped)
```

Comment:
I think `remapped_path_if_available` would be more appropriate here. You should be able to convert a `Path` directly into a `CString` losslessly.

Reply:
done